datavault-common/src/main/java/org/datavaultplatform/common/model/dao/RetrieveDAOImpl.java

```diff
@@ -39,12 +43,15 @@ public void update(Retrieve retrieve) {
 
     @SuppressWarnings("unchecked")
     @Override
-    public List<Retrieve> list() {
+    public List<Retrieve> list(String userId) {
         Session session = this.sessionFactory.openSession();
-        Criteria criteria = session.createCriteria(Retrieve.class);
-        criteria.setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY);
-        criteria.addOrder(Order.asc("timestamp"));
-        List<Retrieve> retrieves = criteria.list();
+        List<Retrieve> retrieves = new ArrayList<>();
+        Optional<Criteria> criteria = retrieveCriteriaForUser(userId, session, Permission.CAN_VIEW_RETRIEVES);
+        if (criteria.isPresent()) {
```

Comment:
FWIW, calling `Optional#isPresent` is usually the wrong approach of dealing with missing values.

E.g.
```java
return retrieveCrtieriaForUser(useId, session, Permission.CAN_VIEW_RETRIEVES)
    .map(criteria -> {
        criteria.setResultTransformer(...);
        criteria.addOrder(...);

        return criteria.list();
    })
    .orElse(Collections.emptyList());
}
```